### PR TITLE
New extension: GHPic

### PR DESCRIPTION
**Pull request by skylark36**
## Description

A simple extension of raycast to upload images to GitHub from clipboard.

## Screencast

https://user-images.githubusercontent.com/19505695/185735238-b0260061-c1f9-45e8-a27c-a953b4b96bcb.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @xiangsanliu 👋

Thanks for your contribution 🔥

I'm not sure what the purpose of this extension is, can you give a/some example(s) to ensure the best possible UI/UX

I tried to test it up through the README but I'm not sure how to. I created a new repository and I'm not sure what to do in order to use this. Can you make describe this in the README?

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by xiangsanliu** [comment]
Hey @pernielsentikaer , thanks for your feedback.
For those who often use markdown to write blogs or articles, it is important to have a stable and easy-to-use image bed if they do not want to save the images locally. The purpose of this extension is to use the GitHub repository as an image bed, quickly upload images, and quickly reference it in the article. I will record a screen later to demonstrate its function.

**Comment by xiangsanliu** [comment]
https://user-images.githubusercontent.com/19505695/186909502-bf31cb8c-228c-474c-a60b-b443fb16c94b.mov

**Comment by xiangsanliu** [comment]
https://user-images.githubusercontent.com/19505695/186912886-ca5c45cf-fec6-4545-a162-86f025a37030.mov

File: extensions/gh-pic/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          title: "Upload failed.",
+        });
+      }
+      setRes(res);
+      setLoaded(true);
+    });
+  }, []);
+
+  return (
+    <List isLoading={loaded}>
```
Comment (pernielsentikaer) [inline extensions/gh-pic/src/index.tsx:118]:
```suggestion
    <List isLoading={!loaded}>
```

File: extensions/gh-pic/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      {res.errorCode == 0 ? (
+        <>
+          <List.Item
+            title={"Copy MarkDown Sytle."}
```
Comment (pernielsentikaer) [inline extensions/gh-pic/src/index.tsx:122]:
```suggestion
            icon={Icon.CopyClipboard}
            title={"Copy as markdown"}
```

File: extensions/gh-pic/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            actions={
+              <ActionPanel>
+                <Action.CopyToClipboard content={`![](${res.picUrl})`} />
+              </ActionPanel>
+            }
+          />
+          <List.Item
+            title={"Copy simple URL."}
```
Comment (pernielsentikaer) [inline extensions/gh-pic/src/index.tsx:130]:
```suggestion
            icon={Icon.CopyClipboard}
            title={"Copy URL"}
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

I was able to set it up and it works as expected. I have some feedback for you to address:

- Is it possible for it to copy a `png file` from Finder? It ends up [like this](https://github.com/pernielsentikaer/pertikaer/blob/main/images/2022-08-26T22:12:39.jpg)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by xiangsanliu** [comment]
Hi, I reproduced this issue:

> Is it possible for it to copy a png file from Finder? It ends up [like this](https://github.com/pernielsentikaer/pertikaer/blob/main/images/2022-08-26T22:12:39.jpg)

I circumvented this problem in code while adding related tips and  Screencast to the README.

> **Note: Due to API restrictions, if you want to upload from a picture file, you need to open the file first and then copy it, instead of copying the file directly in Finder.** see [Upload from file](https://github.com/xiangsanliu/extensions/blob/main/extensions/gh-pic/README.md#upload-from-file)

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**